### PR TITLE
Added "Private" filter option for enhanced product filtering on iOS

### PR DESCRIPTION
**Pull request by pmusolino**
### Summary
This PR introduces the "Private" filter option in the Product List, aligning it with the functionality already present in the Android app. For some reason it was missing. Discussion here: C6H8C3G23/p1717427681770589
This addition addresses user feedback regarding the inability to filter products by their private status.

### Changes
- Added existing "Private" status between the product status filters.
- The product status "Private" now has the description "Private", following the web.
- Just in the product detail, inside `LabeledTextViewTableViewCell` I kept the description "Private published" to follow the same behavior of the web and Android.

### Testing
- Validate the presence of the filter under Product list -> Filter -> Status, and that selecting "Private" selects just private products.
- Validate that when opening a private Product, the tag "Private published" is still there.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/ReusableViews/LabeledTextViewTableViewCell.swift [hunk 1]
```diff
@@ -55,7 +55,16 @@ private extension LabeledTextViewTableViewCell {
     }
 
     func configureProductStatusLabel(productStatus: ProductStatus) {
-        productStatusLabel.text = productStatus.description
+
+        switch productStatus {
+        case .privateStatus:
+            productStatusLabel.text = NSLocalizedString("productDetail.statusLabel",
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/ReusableViews/LabeledTextViewTableViewCell.swift]: non-blocking: should this key be more specific? e.g "productDetail.privateStatusLabel"

**Review by hafizrahman — APPROVED** [review]
This LGTM and works as expected. Leaving a non-blocking question.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/209">19.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/ReusableViews/LabeledTextViewTableViewCell.swift (hunk 1 above)
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/ReusableViews/LabeledTextViewTableViewCell.swift]: Absolutely! Fixed here: c290ff788ece2190d5e333dd819491459ee33c10

**Pull request merged**